native/core/src/execution/datafusion/planner.rs

```diff
@@ -1205,15 +1210,15 @@ impl PhysicalPlanner {
         // to copy the input batch to avoid the data corruption from reusing the input
         // batch.
         let left = if can_reuse_input_batch(&left) {
-            Arc::new(CopyExec::new(left))
+            Arc::new(CopyExec::new(left, CopyMode::UnpackOrDeepCopy))
```

Comment:
Note that `can_reuse_input_batch` returns `true` in fewer cases now, so we are less likely to create the deep-copy version of CopyExec.

Previously we would create a `CopyExec` around any projection or filter, but now we only do so if the projection or filter is wrapping a scan.